src/Runtime/platform.h

```diff
@@ -150,6 +150,11 @@ typedef void (*Event)(void);
 
 #define ARRAY_SIZE(arr) (sizeof(arr) / sizeof((arr)[0]))
 
+#define STATIC_ASSERT(cond)                                                   \
+    struct {                                                                  \
+        int x[1 - 2 * !(cond)];                                               \
+    };
+
```

Comment:
`clang` doesn't like this. Can you explain how it works?

Reply:
I don't think we really need it, since the match checksum enforces that kind of thing anyway.

Reply:
oh, didn't realize clang was used too. in the dark ages before `_Static_assert` it was used to error at compile time for constant expressions: https://elixir.bootlin.com/linux/v4.20.17/source/include/linux/build_bug.h#L66

if `cond` is a compile-time expression that evaluates to `0`, it tries to create an array size of `-1` and errors.

declaring it as anonymous struct allows it to be used at toplevel.

Reply:
yeah I figured there were checks elsewhere, but it was just nice for my build-diff loop to immediately get errors when I miscalculate padding. feel free to remove and I'll just keep it in local stash

Reply:
if you really love it though, probably needs some cpp concat `__FILE__ __LINE__` gensym crap so it can create a unique struct name

Reply:
if you want to keep it, though, should be able to `ifdef` swap it for `_Static_assert`

Reply:
`clang` is used for static analysis (our Issues CI check and editor language servers).

I'm having some internet issues so I'll have to continue this later.

Reply:
After [way too much googling](https://stackoverflow.com/a/54993033) I found something that seems to work.